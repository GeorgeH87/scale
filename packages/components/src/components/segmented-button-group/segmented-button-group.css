/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

:host {
  --background-color: var(--telekom-color-ui-faint);
  --radius: var(--telekom-radius-standard);
  --height: 36px;
  display: flex;
  flex-direction: column;
}

.segmented-button-group {
  background-color: var(--background-color);
  border: 0;
  border-radius: var(--radius);
  padding: 2px;
  width: fit-content;
  height: 32px;
  display: inline-grid;
}

.segmented-button-group--full-width {
  width: 100%;
}

.segmented-button-group--medium {
  height: 40px;
  --height: 40px;
  padding: 0 2px 0 2px;
}

.segmented-button-group--large {
  height: 44px;
  --height: 44px;
  padding: 0 2px 0 2px;
}

.segmented-button-group--label {
  font-size: var(--telekom-typography-font-size-body);
  font-weight: var(--telekom-typography-font-weight-bold);
  margin-bottom: var(--telekom-spacing-unit-x2);
}

.segmented-button-group--helper-text {
  margin-top: var(--telekom-spacing-unit-x2);
}
