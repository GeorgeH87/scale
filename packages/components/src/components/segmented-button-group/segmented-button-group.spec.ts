import { newSpecPage } from '@stencil/core/testing';
import { SegmentedButtonGroup } from './segmented-button-group';

describe('SegmentedButtonGroup', () => {
  it('should match standard snapshot', async () => {
    const page = await newSpecPage({
      components: [SegmentedButtonGroup],
      html: `
        <scale-segmented-button-group>
            <scale-segmented-button>Label</scale-segmented-button>
            <scale-segmented-button>Label</scale-segmented-button>
            <scale-segmented-button>Label</scale-segmented-button>
            <scale-segmented-button>Label</scale-segmented-button>
        </scale-segmented-button-group>`,
    });
    expect(page.root).toMatchSnapshot();
  });
  it('should include right properties', async () => {
    const page = await newSpecPage({
      components: [SegmentedButtonGroup],
      html: `
        <scale-segmented-button-group>
            <scale-segmented-button>Label</scale-segmented-button>
        </scale-segmented-button-group>`,
    });
    page.root.invalid = true;
    page.root.fullWidth = true;
    page.root.helperText = 'helpertext';
    page.root.disabled = true;
    page.root.multiSelect = true;
    page.root.size = 'medium';
    await page.waitForChanges();
    expect(page.rootInstance.invalid).toBe(true);
    expect(page.rootInstance.fullWidth).toBe(true);
    expect(page.rootInstance.helperText).toBe('helpertext');
    expect(page.rootInstance.disabled).toBe(true);
    expect(page.rootInstance.multiSelect).toBe(true);
    expect(page.rootInstance.size).toBe('medium');
  });
  it('should match selected button snapshot', async () => {
    const page = await newSpecPage({
      components: [SegmentedButtonGroup],
      html: `
        <scale-segmented-button-group>
            <scale-segmented-button selected>Label</scale-segmented-button>
            <scale-segmented-button selected>Label</scale-segmented-button>
        </scale-segmented-button-group>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});