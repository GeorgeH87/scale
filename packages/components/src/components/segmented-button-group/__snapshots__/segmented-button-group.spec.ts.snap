// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedButtonGroup should match selected button snapshot 1`] = `
<scale-segmented-button-group>
  <mock:shadow-root>
    <div aria-label="segment button group with 2 buttons" class="segmented-button-group segmented-button-group--small" part="segmented-button-group small" role="group" style="grid-template-columns: repeat(2, 0px);">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <scale-segmented-button adjacent-siblings="right" aria-description-translation="$position $selected" position="1" selected="true" size="small">
    Label
  </scale-segmented-button>
  <scale-segmented-button adjacent-siblings="left" aria-description-translation="$position $selected" position="2" selected="true" size="small">
    Label
  </scale-segmented-button>
</scale-segmented-button-group>
`;

exports[`SegmentedButtonGroup should match standard snapshot 1`] = `
<scale-segmented-button-group>
  <mock:shadow-root>
    <div aria-label="segment button group with 4 buttons" class="segmented-button-group segmented-button-group--small" part="segmented-button-group small" role="group" style="grid-template-columns: repeat(4, 14px);">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <scale-segmented-button adjacent-siblings="" aria-description-translation="$position $selected" position="1" selected="false" size="small">
    Label
  </scale-segmented-button>
  <scale-segmented-button adjacent-siblings="" aria-description-translation="$position $selected" position="2" selected="false" size="small">
    Label
  </scale-segmented-button>
  <scale-segmented-button adjacent-siblings="" aria-description-translation="$position $selected" position="3" selected="false" size="small">
    Label
  </scale-segmented-button>
  <scale-segmented-button adjacent-siblings="" aria-description-translation="$position $selected" position="4" selected="false" size="small">
    Label
  </scale-segmented-button>
</scale-segmented-button-group>
`;